utils/search/search_general.py

```diff
@@ -99,18 +119,41 @@ def search_prepare_parameters(request):
                     }
                     filter_query_split.append(filter)
 
-    if cluster_id != "":  # cluster filter is in a separate query parameter
+    # cluster filter is in a separate query parameter
+    # It facilitates to distinguish between classical and cluster facets.
+    if cluster_id != "":  
         filter_query_split.append({
             'name': "Cluster #" + cluster_id,
             'remove_url': filter_query,
         })
 
+    return filter_query_split
+
+
+def search_prepare_parameters(request):
+    """Construct parameters from the search request object.
```

Comment:
I'd say this description should be a little bit different. Here the idea is that we parse and process all input parameters and aggregate them into a single dict that then is used to build a solr query. What about the first sentence being:

`Parse and pre-process search input parameters from the request object and return them as a dict`